src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/Settings.Cache/SettingsCacheTest.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Core.Suppressions;
+using SonarLint.VisualStudio.Roslyn.Suppressions.Settings.Cache;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Roslyn.Suppressions.UnitTests.Settings.Cache
+{
+    [TestClass]
+    public class SettingsCacheTest
+    {
+        [TestMethod]
+        public void GetSettings_SettingNotInCache_SettingsReadFromFile()
+        {
+            var issues = CreateIssues();
+
+            var cacheObject = CreateCacheObject();
+
+            var fileStorage = new Mock<ISuppressedIssuesFileStorage>();
+            fileStorage.Setup(fs => fs.Get("projectKey")).Returns(issues);
+            
+            var settingsSingleton = CreateSettingsSingleton(cacheObject);
+
+            var testSubject = CreateTestSubject(fileStorage, settingsSingleton);
+            var settings = testSubject.GetSettings("projectKey");
+
+            fileStorage.Verify(fs => fs.Get("projectKey"), Times.Once);
+            cacheObject.ContainsKey("projectKey").Should().BeTrue();
+            cacheObject["projectKey"].Should().BeSameAs(issues);
+            settings.Should().BeSameAs(issues);
+        }
+
+        [TestMethod]
+        public void GetSettings_SettingInCache_SettingsReadFromCache()
+        {
+            var issues = CreateIssues();
+
+            var cacheObject = CreateCacheObject(issues);
+
+            var fileStorage = new Mock<ISuppressedIssuesFileStorage>();
+
+            var settingsSingleton = CreateSettingsSingleton(cacheObject);
+
+            var testSubject = CreateTestSubject(fileStorage, settingsSingleton);
+            var settings = testSubject.GetSettings("projectKey");
```

Comment:
There's a hidden coupling between the `CreateCacheObject` helper method and the tests i.e. that if the helper method returns a pre-populated cache it will contain a single entry with the key `projectKey`. 

One way to make the coupling explicit would be to have explicit helper methods e.g.
```
CreateEmptyCache();
CreatePrePopulatedCache(string key, IEnumerable<SonarQubeIssue> issues);
```

Reply:
After realising CreateEmptyCache() is basicaly one line code It didn't make sense to make it a seperate method.

Reply:
It depends what you are trying to achieve. It doesn't reduce the amount of code, but it does make the test more declarative and easier to understand anyone reading it.